crapssim/bet.py

```diff
@@ -530,6 +533,21 @@ def light_side(self) -> bool:
     def dark_side(self) -> bool:
         return issubclass(self.base_type, (DontPass, DontCome))
 
+    def get_result(self, table: Table) -> BetResult:
+
+        if table.point.status == "Off" and not self.always_working:
+
+            if table.dice.total in (
+                self.get_losing_numbers(table) + self.get_winning_numbers(table)
+            ):
+                # Bet "pushes" and returns to the player
+                return BetResult(
+                    amount=self.amount, remove=True, bet_amount=self.amount
+                )
+                # TODO: should this be self.amount?
+
+        return super().get_result(table)
+
```

Comment:
Main update to odds bet logic that instead pushes when a result would happen (win/loss) on the odds bet if the point is OFF and the `always_working` argument is False